codec/src/main/java/io/netty/handler/codec/compression/Snappy.java

```diff
@@ -565,20 +565,21 @@ private static int decodeCopyWith4ByteOffset(byte tag, ByteBuf in, ByteBuf out,
 
     /**
      * Validates that the offset extracted from a compressed reference is within
-     * the permissible bounds of an offset (4 <= offset <= 32768), and does not
+     * the permissible bounds of an offset (0 < offset < Integer.MAX_VALUE), and does not
```

Comment:
The reason for it limiting to 32768 is as per https://github.com/google/snappy/blob/master/format_description.txt#L79:
> Note that since the current Snappy compressor works in 32 kB blocks and does not do matching across blocks, it will never produce a bitstream with offsets larger than about 32768.

However it then goes on to say
> However, the decompressor should not rely on this, as it may change in the future

So I agree generally that we shouldn't be relying on this but would go on to question the behaviour of other encoders as not being per the 32 kB block chunking spec :)